MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/account/AccountRetrievalService.java

```diff
@@ -0,0 +1,165 @@
+package com.mapbox.mapboxandroiddemo.account;
+
+import android.app.IntentService;
+import android.content.Intent;
+import android.preference.PreferenceManager;
+import android.support.annotation.Nullable;
+import android.util.Base64;
+import android.util.Log;
+
+import com.mapbox.mapboxandroiddemo.MainActivity;
+import com.mapbox.mapboxandroiddemo.model.usermodel.UserResponse;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+
+/**
+ * Background service which retrieves Mapbox Account information
+ */
+
+public class AccountRetrievalService extends IntentService {
+
+  private static final String BASE_URL = "https://api.mapbox.com/api/";
+  private static final String ACCESS_TOKEN_URL = "https://api.mapbox.com/oauth/access_token";
+
+  //  TODO: Fill in CLIENT_SECRET
+  private static final String CLIENT_SECRET = "";
+
+  private static String CLIENT_ID;
+  private static String REDIRECT_URI;
+  private String username;
+
+  public AccountRetrievalService() {
+    super("AccountRetrievalService");
```

Comment:
Are you talking about [these numbers that are passed through to the service via the intent](https://github.com/mapbox/mapbox-android-demo/blob/b2be6f068f7ec7c73f11ba21fa0a012b80aafaf0/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/account/LandingActivity.java#L27-L28)?

If so, they're needed as is because they're part of the oAuth flow. Specifically given to me by our API team.